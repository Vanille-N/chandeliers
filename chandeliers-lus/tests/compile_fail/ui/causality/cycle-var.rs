chandeliers_lus::decl! {
    node foo(i : int) returns (o : int);
    var v, w : int;
    let
        v = o;
        w = i + v;
        o = w;
    tel;
}

fn main() {}
