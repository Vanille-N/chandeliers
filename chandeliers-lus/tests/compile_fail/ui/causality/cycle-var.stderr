error: Variable `w` was found to be part of a dependency cycle
 --> tests/compile_fail/ui/causality/cycle-var.rs:6:9
  |
6 |         w = i + v;
  |         ^
  |
note: The cycle also goes through Variable `o`
 --> tests/compile_fail/ui/causality/cycle-var.rs:7:9
  |
7 |         o = w;
  |         ^
note: The cycle also goes through Variable `v`
 --> tests/compile_fail/ui/causality/cycle-var.rs:5:9
  |
5 |         v = o;
  |         ^
